aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -47,6 +72,22 @@ void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const s
     /**
     * TODO: implement per description 
     */
+
+    if(buffer == NULL || add_entry == NULL)         //return if buffer is NULL or the entry to be added is NULL
+    return;
+
+    if(buffer->in_offs == buffer->out_offs)         //check if the circular buffer is full 
+        buffer->full = true;
+
+    if(buffer->full){                               //if the buffer is full, advance buffer->out_offs to the new start location
+        buffer->out_offs = (buffer->out_offs+1) % (AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED);
+    }
+
+    //add the entry to the ciruclar buffer
+    //NOTE: if the buffer was already full it will overwrite the entry and the read/ out offset is already set to the next pointer
+    buffer->entry[buffer->in_offs] = *add_entry;
+    buffer->in_offs = (buffer->in_offs+1) % (AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED);
```

Comment:
@Mich2899 you could move the check for buffer full at lines 79-80 here here since the buffer is full at this point, not wait until the next call to set it.